mempool/iterators.go

```diff
@@ -156,12 +166,9 @@ func (iter *BlockingIterator) PickLane() types.Lane {
 				iter.cursors[lane].Value.(*mempoolTx).seq == iter.mp.addTxLaneSeqs[lane]) {
 			numEmptyLanes++
 			if numEmptyLanes >= len(iter.sortedLanes) {
-				// There are no lanes with non-accessed entries. Wait until a new tx is added.
-				ch := iter.mp.addTxCh
-				iter.mp.addTxChMtx.RUnlock()
-				<-ch
-				iter.mp.addTxChMtx.RLock()
-				numEmptyLanes = 0
+				// There are no lanes with non-accessed entries. Wait until a
+				// new tx is added.
+				return 0, iter.mp.addTxCh
```

Comment:
Now it returns without advancing the indexes. `pickLane` will be called again starting with the same indexes as when it returned here the previous time.